docs/maintainers/ci.md

```diff
@@ -0,0 +1,134 @@
+  - **sensitive to annotations**
+- Check if all exercise directory names are valid concepts.
+  - **sensitive to annotations**
+
+#### Workflow file
+
+You can install the Concept CI checks for your track by creating a workflow in `.github/workflows/<language>-concept-ci.yml` with the following contents:
+
+```yaml
+name: <language> concept CI
+
+on:
+  push:
+    paths:
+    - 'languages/<language>/config.json'
+    - 'languages/<language>/reference/concepts.csv'
+    - 'languages/<language>/reference/exercise-concepts/**'
+    - 'languages/<language>/exercises/concept/**'
+  pull_request:
+    paths:
+    - 'languages/<language>/config.json'
+    - 'languages/<language>/reference/concepts.csv'
+    - 'languages/<language>/reference/exercise-concepts/**'
+    - 'languages/<language>/exercises/concept/**'
+
+jobs:
+  concept-test:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "Set up Julia"
+        uses: julia-actions/setup-julia@v1
+      
+      - name: Install dependencies
+        run: julia --color=yes --project=.github/bin/concept-checks -e "using Pkg; Pkg.instantiate()"
+
+      - name: Run concept checks
+        run: 
+        run: julia --color=yes --project=.github/bin/concept-checks .github/bin/concept-checks/concept-checks.jl -r . -t <language>
+```
+
+where `<language>` needs to be replaced with the track's language slug.
+
+#### Testing locally
+
+If you want to run the checks locally, you need to [install Julia v1.3][install-julia] and run the following commands:
+
+```
+$ cd languages/<language>/
+$ julia --color=yes --project=../../.github/bin/concept-checks -e "using Pkg; Pkg.instantiate()"
+$ julia --color=yes --project=../../.github/bin/concept-checks ../../.github/bin/concept-checks/concept-checks.jl -t <language>
+```
+
+The second line will install required dependencies and recreate the exact environment the script was developed and tested in, the third will run the checks.
+
+If you don't want to change the working directory, you have to specify the `--root`/`-r` argument pointing to the root of the v3 repository and update the paths to `.github/bin/concept-checks` accordingly.
+
+For convenience, you may want to create symlinks to the script in your track directory:
+
+```
+$ cd languages/<language>/
+$ mkdir bin
+$ cd bin/
+$ ln -s ../../../.github/bin/concept-checks/concept-checks.jl ./concept-checks.jl
+$ ln -s ../../../.github/bin/concept-checks/Project.toml ./Project.toml
+$ ln -s ../../../.github/bin/concept-checks/Manifest.toml ./Manifest.toml
+```
+
+The checks can then be run with
+
+```
+$ julia --project=bin bin/concept-checks.jl -t <language>
+```
+
+from your track directory.
+
+*Note that the script takes a fairly long time before actually running the checks due to compiliation times of the dependencies. This is unfortunately normal when running it as a script. If you're familiar with Julia, you can also `include` it from the REPL to avoid this.*
```

Comment:
This also applies to the CI runs and I really don't like it. I might look into [ahead-of-time compilation](https://github.com/JuliaLang/PackageCompiler.jl) to speed it up in the future, but for now I just wanted it to work.

Reply:
Any idea how slow it is? I don't mind it being a bit slow.

Reply:
The entire run takes about 1:30, the startup time is ~50-60s. Once it's started, it takes a few seconds to run the checks.

Reply:
This is not cached, is it? It would add 1:30 to each PR for languages that have enable this check, right? Would be nice if GH would cache the layers.

Reply:
> Would be nice if GH would cache the layers.

It's more of a Julia problem than a GH problem. That said, PackageCompiler seems a lot more stable now than when I last tried it out, and I was able to reduce the time quite a lot by compiling it ahead-of-time and then caching it.

Reply:
This should be resolved by https://github.com/exercism/v3/pull/718/commits/a29f518d942c55ac79ab97640b985f82e78810ab

It'll take ~5min to build the sysimage once and then it _should_ use the cached build in subsequent runs. I haven't used the actions cache all that much, so I'm not 100% certain that the cache works as intended, but we'll see :D

Reply:
Cool!